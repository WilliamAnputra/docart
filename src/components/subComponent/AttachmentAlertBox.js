import React from 'react';
import {
  View,
  Dimensions,
  Image,
  PixelRatio,
  TouchableOpacity,
  Text
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

import Helper from '../../helper';

import * as globalColors from '../../constants/styles/global';


const SCREEN_HEIGHT = Dimensions.get('window').height;
const SCREEN_WIDTH = Dimensions.get('window').width;

const pixelRatio = PixelRatio.get();

const AttachmentAlertBox = props => {
  return (
    <View style={styles.container}>

      <TouchableOpacity
        style={{ marginTop: 8 }}
        onPress={() => props.onCameraPress()} // callback
      >

        <View style={styles.cameraButton}>

          <Icon name="ios-camera" size={30} color={globalColors.DARK_BLUE} style={{ alignSelf: 'center' }} />
          <Text
            style={styles.actionText}
          >
            Camera
          </Text>

        </View>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => props.onGalleryPress()} // callback
      >


        <View style={styles.galleryButton}>

          <Icon name="ios-albums" size={30} color={globalColors.DARK_BLUE} style={{ alignSelf: 'center' }} />

          <Text
            style={styles.actionText}
          >
            Gallery
          </Text>
        </View>

      </TouchableOpacity>

    </View>
  );
};

const styles = {
  addAttachmentTextContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginTop: Helper.scaleUnit(pixelRatio, 'height', 5)
  },
  actionText: {
    alignSelf: 'center',
    marginLeft: Helper.scaleUnit(pixelRatio, 'width', 25),
  },
  cameraButton: {
    flexDirection: 'row',
    height: Helper.scaleUnit(pixelRatio, 'height', 40),
    width: Helper.scaleUnit(pixelRatio, 'width', 250),
    alignSelf: 'center',
    marginTop: Helper.scaleUnit(pixelRatio, 'height', 40),
    paddingLeft: Helper.scaleUnit(pixelRatio, 'width', 25),
    backgroundColor: globalColors.WHITE
  },
  container: {
    height: Helper.scaleUnit(pixelRatio, 'height', 225),
    width: Helper.scaleUnit(pixelRatio, 'width', 315),
    top: Helper.scaleUnit(pixelRatio, 'height', 100),
    backgroundColor: globalColors.MEDIUM_WHITE,
    borderRadius: 5,
    position: 'absolute',
    alignSelf: 'center',
  },
  galleryButton: {
    flexDirection: 'row',
    height: Helper.scaleUnit(pixelRatio, 'height', 40),
    width: Helper.scaleUnit(pixelRatio, 'width', 250),
    alignSelf: 'center',
    marginTop: Helper.scaleUnit(pixelRatio, 'height', 40),
    paddingLeft: Helper.scaleUnit(pixelRatio, 'width', 25),
    backgroundColor: globalColors.WHITE
  },

  icon: {
    alignSelf: 'center',
    marginLeft: Helper.scaleUnit(pixelRatio, 'width', 9)
  }
};
export { AttachmentAlertBox };
