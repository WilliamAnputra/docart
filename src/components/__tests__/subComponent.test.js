import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import { AttachmentAlertBox } from '../subComponent/AttachmentAlertBox';

it('Renders Attachment screen component', () => {
    const rendered = renderer.create(
        <AttachmentAlertBox>
            <View />
        </AttachmentAlertBox>).toJSON();
    expect(rendered).toMatchSnapshot();
});
