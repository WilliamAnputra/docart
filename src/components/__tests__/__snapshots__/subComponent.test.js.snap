// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Attachment screen component 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#f4f4f4",
      "borderRadius": 5,
      "height": 225,
      "position": "absolute",
      "top": 100,
      "width": 315,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 8,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 40,
          "marginTop": 40,
          "paddingLeft": 25,
          "width": 250,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#253f61",
              "fontSize": 30,
            },
            Object {
              "alignSelf": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "marginLeft": 25,
          }
        }
      >
        Camera
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 40,
          "marginTop": 40,
          "paddingLeft": 25,
          "width": 250,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#253f61",
              "fontSize": 30,
            },
            Object {
              "alignSelf": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "marginLeft": 25,
          }
        }
      >
        Gallery
      </Text>
    </View>
  </View>
</View>
`;
